src/System.Private.ServiceModel/tools/scripts/WcfTestServerSetup.cmd

```diff
@@ -0,0 +1,191 @@
+if ERRORLEVEL 1 goto :Failure
+
+:: Setup PR service only if this is a master repo
+:: Uncomment line below when we switch to use one central PRService
+REM if /I NOT '%MasterRepo%'=='%CurrentRepo%' goto :SkipPRService
+
+:: Create a new application pool and an application for the PR service
+echo Create IIS applicaton pool: %PRServiceName%
+call :Run %appcmd% add apppool /name:%PRServiceName% /managedRuntimeVersion:v4.0 /managedPipelineMode:Integrated
+if ERRORLEVEL 1 goto :Failure
+echo Create IIS applicatoin: %PRServiceName%
+call :Run %appcmd% add app /site.name:"Default Web Site" /path:/%PRServiceName% /physicalPath:%CurrentRepo%\src\System.Private.ServiceModel\tools\PRService -applicationPool:%PRServiceName%
+if ERRORLEVEL 1 goto :Failure
+
+:: Grant app pool %PRServiceName% "Modify" access to %CurrentRepo% and its subdirectories
+echo Grant app pool %PRServiceName% "Modify" access to %CurrentRepo% and its subdirectories
+call :Run icacls %CurrentRepo% /grant:r "IIS APPPOOL\%PRServiceName%":M /T /Q
+if ERRORLEVEL 1 goto :Failure
+:: Uncomment below and delete above when we switch to central PR service
+REM echo Grant app pool %PRServiceName% "Read and Execute" access to the PRService and its subdirectories
+REM call :Run icacls %CurrentRepo%\src\System.Private.ServiceModel\tools\PRService /grant:r "IIS APPPOOL\%PRServiceName%":RX /T /Q
+REM if ERRORLEVEL 1 goto :Failure
+
+:SkipPRService
+
+:: Grant master PR service's app pool access to current repo if master PRService exists
+if /I '%MasterRepo%'=='%CurrentRepo%' goto :SkipGrantAccess
+if NOT EXIST %MasterRepo% goto :SkipGrantAccess
+%appcmd% list apppool %PRServiceMaster% >nul
+if ERRORLEVEL 1 goto :SkipGrantAccess
+echo Grant master PRServer app pool: %PRServiceMaster% "Modify" access to %CurrentRepo% and its sudirectories
+call :Run icacls %CurrentRepo% /grant:r "IIS APPPOOL\%PRServiceMaster%":M /T /Q
+if ERRORLEVEL 1 goto :Failure
+
+:SkipGrantAccess
+
+:: Install certificates if %WcfTestDir% does not exist; otherwise, assume certs are installed already
+if EXIST %WcfTestDir% goto :SkipCertInstall
+
+:: Make sure there is an HTTPS binding in IIS site "Default Web Site"
+%appcmd% list site "Default Web Site" /text:* | findstr /i bindings | findstr /i https >nul
+if ERRORLEVEL 1 (
+    echo Add an HTTPS binding to "Default Web Site"
+    call :Run %appcmd% set site "Default Web Site" /+bindings.[protocol='https',bindingInformation='*:443:']
+    if ERRORLEVEL 1 goto :Failure
+)
+
+:: Use current repo for cert gen if master repo does not exist
+if NOT EXIST %MasterRepo% (
+    set MasterRepo=%CurrentRepo%
+    set WcfServiceMaster=%WcfServiceName%
+)
+
+echo Build CertificateGenerator tool
+call :Run %MasterRepo%\src\System.Private.ServiceModel\tools\scripts\BuildCertUtil.cmd
+if ERRORLEVEL 1 goto :Failure
+
+echo Run CertificateGenerator tool. This will take a little while...
+md %WcfTestDir%
+set certGen=%MasterRepo%\bin\Wcf\tools\CertificateGenerator\CertificateGenerator.exe
```

Comment:
Nit, not blocking -- like Jason I like to see env vars set at the top.  Easier to see and maintain.